adagrams/game.py

```diff
@@ -1,11 +1,99 @@
+import random
+
 def draw_letters():
-    pass
+
+    letter_distribution = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+    ten_rand_letters = []
+
+    while len(ten_rand_letters) < 10:
+        random_letter =random.choice(list(letter_distribution.keys()))   
+        
+        if ten_rand_letters.count(random_letter) < letter_distribution[random_letter]:
+            ten_rand_letters.append(random_letter)
+    return ten_rand_letters
 
 def uses_available_letters(word, letter_bank):
-    pass
+
+    upper_word = word.upper()
+    
+    for letter in upper_word:
+        if letter not in letter_bank or letter_bank.count(letter) < upper_word.count(letter):
```

Comment:
As above, be careful using `count` in a loop, since it is also a loop. Again, because of the small size of the hands we're working with, we won't see a performance issue, but in a more general case, we should be aware of whether we have extraneous nested loops.

In this case, we could build a dictionary of the counts of the letters in the word, and another holding the counts of the letters in the hand. Then we could compare the count of each word letter to the count of each hand letter (like here, but we would have counted the word and hand letters only once rather than each time through the loop).